hyperspy/signal.py

```diff
@@ -4794,6 +4806,78 @@ def T(self):
         """
         return self.transpose()
 
+    def apply_apodization(self, type='hann', hann_order=None, tukey_alpha=0.5, inplace=False):
+        """
+        Apply apodization window to a signal either in place or return a new signal.
+
+        Parameters
+        ----------
+        type : string, optional
+            Select between 'hann', 'hamming', 'tukey'
+            (Default: 'hann')
+        hann_order : None or int, optional
```

Comment:
What does hann_order=None mean if hann window is used? 

Also, add a line of "only used if `window='hann'`